src/deploy/hosting/release.ts

```diff
@@ -15,38 +14,40 @@ export async function release(context: Context, options: Options, payload: Paylo
     return;
   }
 
-  const projectNumber = await needProjectNumber(options);
-
   logger.debug(JSON.stringify(context.hosting.deploys, null, 2));
   await Promise.all(
     context.hosting.deploys.map(async (deploy) => {
       utils.logLabeledBullet(`hosting[${deploy.site}]`, "finalizing version...");
 
-      const config = await convertConfig(context, payload, deploy.config, true);
-      const data = { status: "FINALIZED", config };
-      const queryParams = { updateMask: "status,config" };
-
-      const finalizeResult = await client.patch(`/${deploy.version!}`, data, { queryParams });
-
-      logger.debug(`[hosting] finalized version for ${deploy.site}:${finalizeResult.body}`);
+      const update: Partial<api.Version> = {
+        status: "FINALIZED",
+        config: await convertConfig(context, payload, deploy.config, /* finalize= */ true),
+      };
+
+      const parts = deploy.version!.split("/");
+      const versionId = parts[parts.length - 1];
+      const finalizedVersion = await api.updateVersion(deploy.site, versionId, update);
+
+      logger.debug(
+        `[hosting] finalized version for ${deploy.site}:${JSON.stringify(
+          finalizedVersion,
+          null,
+          2
+        )}`
```

Comment:
I'd prefer we don't bloat the debug lines with newlines if we can help it. It makes it impossible to get the entire log by just looking for a single line (i.e. `cat log | grep "some request"` doesn't work - the rest of the object is on new lines)